File: 0204-count-primes/solution.py
import math
class Solution:        
    def countPrimes(self, n: int) -> int:
        if n <= 2:
            return 0
        elif n == 3:
            return 1
        elif n == 4 or n == 5:
            return 2
        elif n == 6:
            return 3

        is_Prime = [True]*n
        is_Prime[0] = False
        is_Prime[1] = False
        
        i = 2
        while i*i < n:
            if is_Prime[i]:
                j = i*i
                while j < n:
                    is_Prime[j] = False
                    j += i
            i += 1
    
        count = 0
        for i in range(len(is_Prime)):
            if is_Prime[i]:
                count += 1
        return count

File: 0204-count-primes/solution.cpp
class Solution {
public:
    bool checkPrime(int num)
    {
        if(num == 2 || num == 3)
            return true;
        for(int i = 2; i*i <= num; i++)
        {
            if(num%i == 0)
                return false;
        }
        
        return true;
    }
    int countPrimes(int n) 
    {
        if(n == 0 || n == 1)
            return 0;

        n--;
        const int size = n+1;
        vector<bool>isPrime(size, true);
        isPrime[0] = isPrime[1] = false;
        
        for(int num = 2; num*num <= n; num++)
        {
            if(isPrime[num])
            {
                for(int multiple = num*num; multiple <= n; multiple += num)
                {
                    isPrime[multiple] = false;
                }
            }
        }
        
        int numPrime = 0;
        for(int i = 2; i < n+1; i++)
        {
            numPrime += isPrime[i];
        }
        return numPrime;
        
    }
};
